addons/docs/src/blocks/Story.tsx

```diff
@@ -105,6 +106,15 @@ const Story: FunctionComponent<StoryProps> = (props) => {
   const ref = useRef();
   const story = useStory(getStoryId(props, context), context);
 
+  // Ensure we wait until this story is properly rendered in the docs context.
+  // The purpose of this is to ensure that that the `DOCS_RENDERED` event isn't emitted
+  // until all stories on the page have rendered.
+  const { id: storyId, registerRenderingStory } = context;
+  const storyRendered = useMemo(registerRenderingStory, [storyId]);
+  useEffect(() => {
```

Comment:
does it matter what order the `useEffect`s are executed in? are there any race conditions?

Reply:
I think we need to think more about the emitting behaviour we want from v7 inline rendering (should it emit `STORY_RENDERED`? What should the hooks wait for?)

So we will revisit this. I don't think there are race conditions, they just execute in order.